spec/controllers/whitehall_assets_controller_spec.rb

```diff
@@ -94,6 +94,17 @@
         expect(response).to have_http_status(:unprocessable_entity)
       end
     end
+
+    context "an asset with the same legacy_url_path as an existing asset" do
+      let(:attributes) { FactoryBot.attributes_for(:whitehall_asset, :with_legacy_metadata) }
+      let!(:existing_asset) { FactoryBot.create(:whitehall_asset, legacy_url_path: attributes[:legacy_url_path]) }
+
+      it "marks the existing asset as deleted" do
+        post :create, params: { asset: attributes }
+
+        expect(existing_asset.reload).to be_deleted
```

Comment:
Observation: I'm slightly surprised that reloading a deleted asset works - I'd have thought it might result in a `RecordNotFound` exception.

Reply:
Yes, it's slightly odd. This pattern is used in the [mongoid_paranoia](https://github.com/simi/mongoid_paranoia/blob/e0c6608daa61c8ecf1165e278565b1b6da1f2671/spec/mongoid/nested_attributes_spec.rb#L98) specs too.